### PR TITLE
Fixed codecvt deprecation warning

### PR DESCRIPTION
**Pull request by biodranik**

### CODE REVIEW

File: indexer/editable_map_object.cpp [hunk 1]
```diff
@@ -743,38 +742,30 @@ bool EditableMapObject::ValidateName(string const & name)
...
+    // Emoji modifiers https://en.wikipedia.org/wiki/Emoji#Emoji_versus_text_presentation
+    if (ch == U'\U0000FE0E' || ch == U'\U0000FE0F')
+      return false;
     // Exclude format controls, musical symbols, emoticons, ornamental and pictographs,
     // ancient and exotic alphabets.
     if (ch >= U'\U0000FFF0' && ch <= U'\U0001F9FF')
       return false;
 
-    if (find(excludedSymbols.cbegin(), excludedSymbols.cend(), ch) != excludedSymbols.cend())
+    if (excludedSymbols.find(ch) != excludedSymbols.npos)
```
Comment (AndrewShkrob) [inline indexer/editable_map_object.cpp:768]: Clang-Tidy: Static member accessed through instance 

```suggestion
    if (excludedSymbols.find(ch) != std::u32string_view::npos)
```

or

```suggestion
    if (excludedSymbols.find(ch) != std::u32string::npos)
```

Comment (biodranik) [reply indexer/editable_map_object.cpp]: godbolt doesn't show any difference in the generated code, so Clang-Tidy's warning is not critical. Fixed it.

File: indexer/editable_map_object.cpp [hunk 2]
```diff
@@ -743,38 +742,30 @@ bool EditableMapObject::ValidateName(string const & name)
...
-
-  std::u32string const excludedSymbols = U"^~§><{}*=_±\n\t\r\v\f|√•÷×¶";
-
-  for (auto const ch : u32name)
+  using Iter = utf8::unchecked::iterator<string::const_iterator>;
+  for (Iter it{name.cbegin()}; it != Iter{name.cend()}; ++it)
   {
+    auto const ch = *it;
+    // Exclude ASCII control characters.
+    if (ch < U'\U0000001F')
```
Comment (vng) [inline indexer/editable_map_object.cpp:752]: nit: Why so complicated constants instead of simple 0x1F, 0x1F9FF, .. here and below?

**Review by vng — APPROVED** [review]

File: indexer/editable_map_object.cpp (hunk 2 above)
Comment (AndrewShkrob) [reply indexer/editable_map_object.cpp:752]: `U'\U1F'` -> error: incomplete universal character name

Comment (biodranik) [reply indexer/editable_map_object.cpp:752]:
```suggestion
    if (ch <= U'\U0000001F')
```

Comment (vng) [reply indexer/editable_map_object.cpp]: ? Just ```if(ch <= 0x1F)```

**Pull request merged**